Migration from Gtk3 to Gtk4 and libadwaita 1.7

Scroker opened the pull request:
This pull request collects the changes I made in order to move the UI of the application from gtk3 to gtk4 and libadwaita 1.7. The version of gnome that is compatible to date is 48. 
In general I tried to reuse as much code as possible but obviously the js and xml files related to the UI component are deeply changed. On the daemon instead I intervened almost exclusively to make it compatible with Gtk4. 
Finally I reorganized the icon part of the gresource to natively get an Icon Theme.

ferdnyc committed:
WIP: No-cache SMS support

Scroker committed:
Initial mock of in GTK4

Scroker committed:
Now the deamon works

Scroker committed:
New graphics, sendfile now works

Scroker committed:
Added the pairing function

Scroker committed:
Removed some non system icon

Scroker committed:
some graphical improvements to the pair ui

Scroker committed:
A little bugfixing

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Add extension and app button to homepage

Scroker committed:
Style improvements to the home page

Scroker committed:
Clean de device.js code

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Merge branch 'main' into gtk4-port

LorierMatthieu committed:
Switch messaging-window.ui to Gtk4/Adwaita

Scroker committed:
Migrate the findmyphone deprecated messagedialog component to a notification

Scroker committed:
Merge branch 'gtk4-port' of github.com:Scroker/gnome-shell-extension-gsconnect into gtk4-port

LorierMatthieu committed:
Update messaging-window.ui

LorierMatthieu committed:
Update messaging-window.ui

Scroker committed:
Start the migration of messaging.js to the new UI

LorierMatthieu committed:
Switch service-error-dialog.ui to Gtk4/Adwaita

LorierMatthieu committed:
Switch mousepad-input-dialog.ui to Gtk4/Adwaita

LorierMatthieu committed:
Update mousepad-input-dialog.ui

LorierMatthieu committed:
Move screenshots to a new folder and adapt files

LorierMatthieu committed:
Switch messaging-conversation.ui to Gtk4/Adwaita

LorierMatthieu committed:
Update messaging-conversation.ui to use second-icon

LorierMatthieu committed:
Switch messaging-conversation-summary.ui to Gtk4/Adwaita

LorierMatthieu committed:
Update service-error-dialog.ui

LorierMatthieu committed:
Switch messaging-conversation-message.ui to Gtk4/Adwaita

Scroker committed:
Switch the code to tne new messaging-window.ui

Scroker committed:
Added icons button to messages conversation

LorierMatthieu committed:
Update preferences-window.ui design (no JavaScript changes needed)

Scroker committed:
Merge branch 'gtk4-port' of github.com:Scroker/gnome-shell-extension-gsconnect into gtk4-port

Scroker committed:
I made some changes to the messages ui and removed some logs

Scroker committed:
Merge branch 'no-cache-sms' of github.com:ferdnyc/gnome-shell-extension-gsconnect into ferdnyc-no-cache-sms

Scroker committed:
Migrated the conversation from stack to AdwNavigationPage

Scroker committed:
Contact chooser UI migrated to gtk4

Scroker committed:
Merge branch 'ferdnyc-no-cache-sms' into gtk4-port

LorierMatthieu committed:
Update connect-dialog.ui design

LorierMatthieu committed:
Switch service-device-chooser.ui to Gtk4/Adwaita

Scroker committed:
Rollback from PreferencePage to ToolbarView on connect-dialog.ui

LorierMatthieu committed:
Update connect-dialog.ui file

LorierMatthieu committed:
Update command-row.ui file

LorierMatthieu committed:
Update command-row.ui file

LorierMatthieu committed:
Update contact-chooser.ui file

LorierMatthieu committed:
Update contacts-adress-row.ui file

LorierMatthieu committed:
Update command-row.ui file

LorierMatthieu committed:
Update device-pair.ui file

LorierMatthieu committed:
Add a commentary on device-pair.ui file

LorierMatthieu committed:
Update messaging-conversation.ui file

LorierMatthieu committed:
Update messaging-conversation-message.ui file

LorierMatthieu committed:
Update some UI files

LorierMatthieu committed:
Update messaging-conversation.ui file

LorierMatthieu committed:
Update messaging-conversation-summary.ui file

LorierMatthieu committed:
Update messaging-window.ui file

LorierMatthieu committed:
Update preferences-command-editor.ui file

LorierMatthieu committed:
Update preferences-section-row.ui file

Scroker committed:
Some code improovements to the message window

Scroker committed:
Merge branch 'gtk4-port' of github.com:Scroker/gnome-shell-extension-gsconnect into gtk4-port

Scroker committed:
Rollback on the messaging-conversation-summary.ui and invert counter and time as in gtk3 interface

LorierMatthieu committed:
Various improvements to UI files

Scroker committed:
Added the EventController on ContactChooser

Scroker committed:
Added the EventController on ContactChooser

Scroker committed:
Added BottomSheet on mousepad dialog and ToggleGroup on volume

Scroker committed:
Restore active when screen is locked option

Scroker committed:
Added to the device-page.ui the experimental plugin rows

Scroker committed:
Some UI improovements and bugfixing

Scroker committed:
I made a GtkListBox automatically built from the GMenuModel of the device

Scroker committed:
I made a GtkListBox automatically built from the GMenuModel of the device

LorierMatthieu committed:
Mark a string as translatable

I managed to commit from my phone.

Scroker committed:
Now the device preference page show the commands menu

Scroker committed:
Porting of shortcut editor to AdwDialog

Scroker committed:
Porting of shortcut editor to AdwDialog

Scroker committed:
Porting of shortcut editor to AdwDialog

Scroker committed:
Porting of shortcut editor to AdwDialog

Scroker committed:
Fixed find-my-phone bugs, igrated to adw the ui and added a notification

Scroker committed:
Added a popup to show the other partecipants on a chat

Scroker committed:
Added a popup to show the other partecipants on a chat

Scroker committed:
Mousepad needs some improovements but works!

Scroker committed:
Created a new sharesms ui and ported service-device-chooser and service-error-dialog

Scroker committed:
New icons for messages and ring

Scroker committed:
The extension image now fits the link button

Scroker committed:
Now legacy message should be work

Scroker committed:
Some improvements to legacy-messagjng-dialog.ui

Scroker committed:
Now also the notification plugin should be work

Scroker committed:
Some bugfixing on new notification plugin

Scroker committed:
Fixed clipboard plugin, now it works again

Scroker committed:
Removed some debug logs

Scroker committed:
Removed some debug logs

Scroker committed:
Some changes to make themed icons in the app without code implementation

Scroker committed:
Italian traanslations added and title removed

Scroker committed:
Now the gtk theme works withi user and system installs

Scroker committed:
Mooved some option from the menu to a new PreferenceDialog

Scroker committed:
The device name row now works

Scroker committed:
General bugfixing

Scroker committed:
General Bugfixing

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Adding comment and general bugfixing

Scroker committed:
Updates to the icon theme and comments added to service.js

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
First change correction before pull request

Scroker committed:
Second change correction before pull request

Scroker committed:
Set compatibility with Gnome Shell

Scroker committed:
Fixed translations and stylesheet

Scroker committed:
Restored accessibility

Scroker committed:
UI Changes and documentation

LorierMatthieu committed:
Minor improvements and update french translation

Scroker committed:
Updated translation using gettext

Scroker committed:
Merge branch 'gtk4-port' of github.com:Scroker/gnome-shell-extension-gsconnect into gtk4-port

Scroker committed:
Add gettext-domain on metadata

LorierMatthieu committed:
Make messaging-conversation.ui take all space (width)

LorierMatthieu committed:
Mark a string as translatable

LorierMatthieu committed:
Adding new screenshots to reflect new design

LorierMatthieu commented:
I can’t wait for the fork to be pushed. Congratulations, Giorgio!

Scroker committed:
New message text input component created

Scroker committed:
Add searchbutton to the threads

LorierMatthieu committed:
Updating screenshots

Scroker committed:
Add GtkEventoControllerKey also on legacyMessageing.js

Scroker committed:
Merge branch 'gtk4-port' of github.com:Scroker/gnome-shell-extension-gsconnect into gtk4-port

Scroker committed:
Added our credits under designer in the about page

Scroker committed:
Fix offset problem on entering in the mousepad label surface

Scroker committed:
Set the minimum window size to smartphone screen

Scroker committed:
Fix double chars on mousepad and created right click gesture

ferdnyc commented:
There are several _hundred_ eslint errors coming up with the code in this PR.

@Scroker if you go to the root of your worktree and run `npm install` followed by `npm install eslint`, you can use `npx eslint --fix` to correct probably most of them.

ferdnyc commented:
> @Scroker if you go to the root of your worktree and run `npm install` followed by `npm install eslint`, you can use `npx eslint --fix` to correct probably most of them.

`npx eslint --fix-dry-run` (which simulates running eslint with its `--fix` changes applied) drops the count down to "✖ 73 problems (43 errors, 30 warnings)" so it's definitely a good start.

Scroker committed:
List of fixed item:
- ESlint test failure.
- Some of the problems during the destruction of GTKWidget.
- Service error window now works

ferdnyc commented:
(Note, also, that there are a number of docstring issues that I suspect `--fix` may fix **wwrong**, because they're actually being misinterpreted by it due to wrong tags being used — mostly `@type` which isn't really meant for function docstrings.

The remaining issues that can't be autofixed are:

> src/preferences/device.js
>    294:21  error    _init() that only calls super._init() is unnecessary                      no-restricted-syntax

Like it says, if the only point of an `_init` override is to call `super._init()`, just don't override the init and let it inherit.

>    632:19  error    'hasBattery' is assigned a value but never used                           no-unused-vars

Yeah, I don't get why this check is there. It appears to do nothing.

>    653:34  error    'e' is defined but never used                                             no-unused-vars

Don't use `} catch (e) {` unless you're going to do something with the error. Otherwise, just `} catch {`.

>    845:1   warning  Expected @param names to be "name". Got "settings, key, name"             jsdoc/check-param-names
>    848:1   warning  Missing JSDoc @param "name" description                                   jsdoc/require-param-description
>    848:1   warning  Missing JSDoc @param "name" type                                          jsdoc/require-param-type

Because that entire docblock was copy-pasted from the function above. Though, confusingly, it contains _more_ content for some of the incorrect parameters, than the docblock for the function that actually uses those parameters.

>    952:37  error    'labelGtk' is not defined                                                 no-undef

Pretty sure that's just a typo, `labelGtk.accelerator_get_label` should just be `Gtk.accelerator_get_label`.

>   1032:22  error    'e' is defined but never used                                             no-unused-vars
>   1091:18  error    'e' is defined but never used                                             no-unused-vars
>   1162:1   warning  Expected @param names to be "settings, key". Got "widget, settings, key"  jsdoc/check-param-names
>   1163:1   warning  Missing JSDoc @param "settings" description                               jsdoc/require-param-description
>   1163:1   warning  Missing JSDoc @param "settings" type                                      jsdoc/require-param-type
>   1164:1   warning  Missing JSDoc @param "key" description                                    jsdoc/require-param-description
>   1164:1   warning  Missing JSDoc @param "key" type                                           jsdoc/require-param-type
>   1166:5   error    Duplicate name '_onReceiveDirectoryChanged'                               no-dupe-class-members

That entire function appears twice, with the only difference being that the second copy sets the subtitle of the dialog to `receiveDir`. (And has a completely unrelated docblock above it.) The first copy does this, which is confusing:

```js
        if (this.receive_directory.get_subtitle() !== receiveDir)
            this.receive_directory.set_subtitle();
```

> src/preferences/service.js
>   498:1   warning  Missing JSDoc @param "widget" description                                       jsdoc/require-param-description
>   498:1   warning  Missing JSDoc @param "widget" type                                              jsdoc/require-param-type

\*nod\* Just plain missing.

>   598:25  error    Expected '===' and instead saw '=='                                             eqeqeq
>   611:35  error    Expected '===' and instead saw '=='                                             eqeqeq
>   726:27  error    Expected '===' and instead saw '=='                                             eqeqeq
>   776:49  error    Expected '!==' and instead saw '!='                                             eqeqeq
>   777:59  error    Expected '===' and instead saw '=='                                             eqeqeq

Please maintain strict comparison discipline; triple-equals and friends are _your_ friends.

>   786:1   warning  Expected @param names to be "service, device". Got "box, row, service, device"  jsdoc/check-param-names
>   789:1   warning  Missing JSDoc @param "service" description                                      jsdoc/require-param-description
>   789:1   warning  Missing JSDoc @param "service" type                                             jsdoc/require-param-type
>   790:1   warning  Missing JSDoc @param "device" description                                       jsdoc/require-param-description
>   790:1   warning  Missing JSDoc @param "device" type                                              jsdoc/require-param-type

This is due to what looks like a copy-paste error, the docblock here is:

```js
    /**
     * Updates the UI based on the service's active state. If the service is active,
    /**
     * Updates the UI based on the service's active state. If the service is active,
     * it sets the window subtitle to "Searching for devices" and enables the refresh button.
     * If the service is inactive, it sets the subtitle to "Waiting for service" and disables
     * the refresh button.
     *
     * @param {object} service - The service object whose state changed.
     * @param {object} pspec - The property specification object.
     * 
     * @returns {void}
     */
```

So the "first" one (which is only one line long) gets interpreted wrong.

>   796:21  error    Expected '!==' and instead saw '!='                                             eqeqeq
> 
> src/service/components/contacts.js
>   375:23  error  Expected '===' and instead saw '=='  eqeqeq
> 
> src/service/plugins/share.js
>   272:22  error  'e' is defined but never used                           no-unused-vars
>   389:7   error  'FileChooserDialog' is assigned a value but never used  no-unused-vars

If that class is no longer needed (and it looks like it's not), it can just be deleted.

> 
> src/service/plugins/sms.js
>   289:13  error    Closing curly brace does not appear on the same line as the subsequent block  @stylistic/js/brace-style

eslint wants this style:

```js
if () {
  ...
  ...
} else {
  ...
}
```

not: 

```js
if () {
  ...
  ...
}

else {
  ...
}
```

...And I'm inclined to agree. Having the "detached else" is confusing and easy to misread.

>   304:1   warning  Missing JSDoc @param "numberToGet" description                                jsdoc/require-param-description
>   304:1   warning  Missing JSDoc @param "numberToGet" type                                       jsdoc/require-param-type
>   305:1   warning  Missing JSDoc @param "earliestTimestamp" description                          jsdoc/require-param-description
>   305:1   warning  Missing JSDoc @param "earliestTimestamp" type                                 jsdoc/require-param-type

Yeah, that's kinda my fault, sorry. Never got around to documenting those.

>   337:1   warning  Expected @param names to be "sender, body". Got "hint, sender, body"          jsdoc/check-param-names
>   338:1   warning  Missing JSDoc @param "sender" description                                     jsdoc/require-param-description
>   338:1   warning  Missing JSDoc @param "sender" type                                            jsdoc/require-param-type
>   339:1   warning  Missing JSDoc @param "body" description                                       jsdoc/require-param-description
>   339:1   warning  Missing JSDoc @param "body" type                                              jsdoc/require-param-type

Docs need updating for function changes.
 
> src/service/ui/components.js
>   70:23  error  Expected '===' and instead saw '=='  eqeqeq
> 
> src/service/ui/contacts.js
>   132:1  warning  The type 'number-selected' is undefined  jsdoc/no-undefined-types

`@emits` isn't really meant for GLib signals, it's interpreted by JSDoc as a JavaScript event. However, you can probably force it to take the argument by wrapping it in single quotes, i.e. `@emits 'number-selected'`

> src/service/ui/legacyMessaging.js
>     8:9  error  'MessagingInputText' is defined but never used                         no-unused-vars

Yeah, if these imports are needed for the template, you can make eslint happy by adding a preceding comment like,

```
// eslint-disable-next-line no-unused-vars -- needed for ui builder
```

(Please don't place long trailing comments at the end of already long lines, ideally every source line should try to keep within 80 columns of length. Sometimes exceeding that is inevitable, but the _worst_ reason to exceed it is a trailing comment that should really be on the previous line, instead.)

>   178:9  error  Expected an assignment or function call and instead saw an expression  no-unused-expressions

Line ends with a comma, should be a semicolon.
> 
> src/service/ui/messaging.js
>     12:9   error    'MessagingInputText' is defined but never used  no-unused-vars
>     19:7   error    'Tweener' is assigned a value but never used    no-unused-vars

>    193:10  error    'setAvatarVisible' is defined but never used    no-unused-vars

Here again, all calls to this function were removed, so if it's not needed anymore it can just be dumped.

>    210:1   warning  Missing JSDoc @returns declaration              jsdoc/require-returns
>    212:1   warning  Missing JSDoc @param "phone" description        jsdoc/require-param-description
>    212:1   warning  Missing JSDoc @param "phone" type               jsdoc/require-param-type

>    294:33  error    '_len' is assigned a value but never used       no-unused-vars

You can use just a single underscore (`_`) to indicate an unused value in a destructuring assignment.

>    296:28  error    'GtkPixbuf' is not defined                      no-undef

That should be `GdkPixbuf`, but it's not imported either.

>    304:24  error    'loader' is not defined                         no-undef

If you want to define `loader` for use outside the try block, you'll need to hoist it. Declare it with `let` before the `try`, then assign to the already-declared variable inside the block.

>    568:37  error    Expected '===' and instead saw '=='             eqeqeq
>    815:20  error    Expected '!==' and instead saw '!='             eqeqeq
>    815:46  error    Expected '!==' and instead saw '!='             eqeqeq
>    974:26  error    Expected '===' and instead saw '=='             eqeqeq
>   1146:27  error    Expected '===' and instead saw '=='             eqeqeq
>   1273:1   warning  The type 'AddressRow' is undefined              jsdoc/no-undefined-types

Defined in `src/service/ui/contacts.js` but not exported; if you want to use it in `messaging.js` you'll need to change it to `export` and refer to it as `Contacts.AddressRow`.

>   1459:26  error    Expected '===' and instead saw '=='             eqeqeq
>   1474:21  error    'old_thread_id' is constant                     no-const-assign

Needs to be declared `let` if you're going to reassign it.

>   1544:9   error    'scrolledWindow' is not defined                 no-undef
>   1562:44  error    'message' is not defined                        no-undef

Yeah, I don't get the changes here at all. Code was deleted that makes both of those undefined, but they're still used.

> src/service/ui/mousepad.js
>    83:29  error  'offset_x' is assigned a value but never used  no-unused-vars
>    83:39  error  'offset_y' is assigned a value but never used  no-unused-vars

If you don't need the extra parameters to the function, just leave 'em out of the signature.

>   203:13  error  'keyvalLower' is constant                      no-const-assign
>   209:13  error  'keyvalLower' is constant                      no-const-assign

It used to be declared with `let` for a reason.

> src/service/ui/notification.js
>   8:9  error  'MessagingInputText' is defined but never used  no-unused-vars
> 
> src/service/ui/service.js
>    98:1   warning  The type 'Integer' is undefined                            jsdoc/no-undefined-types

JS uses 'number'.

>   114:5   error    Expected getter 'response' to be before setter 'response'  grouped-accessor-pairs
>   205:49  error    'error' is not defined                                     no-undef

The assignment of `this.error` was removed.

> 
> src/service/utils/ui.js
>   5:8  error  'Gdk' is defined but never used  no-unused-vars

No longer needed, all references were removed.
 
> ✖ 73 problems (43 errors, 30 warnings)

Scroker commented:
Thanks @ferdnyc , I should have corrected all the errors and warnings. During this process I also realized that:
- The dialog box for reporting problems was not working and I fixed it.
- There are some errors in the gjs console when shutting down the daemon, I fixed them partially, but from what I understand the strategy to destroy GtkWidget in Gtk4 has changed.

ferdnyc commented on src/preferences/device.js:
```diff
@@ -1286,18 +1251,19 @@ export const DeviceNavigationPage = GObject.registerClass({
      */
-    dispose() {
+    /*
+    vfunnc_finalize() {
```
It feels like you're flailing here. Renaming `dispose()` to `vfunc_finalize()` (spelled wrong), and commenting it out, all in the same commit with lots of other changes... There's no clear justification for doing this, and there's no way I'm going to merge a commented-out, misspelled function without assurances that it's no longer needed to perform any of the very necessary-sounding cleanup tasks listed in its docstring.

Scroker replied:
Well, here I really need a hand to understand. In fact I was about to write to you. 
I've been researching all night and, correct me if I'm wrong, it seems that in Gtk4 the destroy signal and vfunc_destroy are no longer available. 
The impression is that they encourage the use of the garbage collector and the avoidance of invoking destroy directly. Only  for the gtk widget. The other GObject works normally.
It's right?

ferdnyc commented on src/service/ui/legacyMessaging.js:
```diff
@@ -5,7 +5,7 @@
 import Gtk from 'gi://Gtk?version=4.0';
 import Adw from 'gi://Adw';
-import { MessagingInputText } from './components.js'; // this dependency is needed for template build
+import {MessagingInputText} from './components.js'; // this dependency is needed for template build
```
OK, but with these changes that trailing comment is no longer accurate and should be removed. Same for the other instances.

Scroker replied:
Ok I need to review each comment.

ferdnyc reviewed (changes requested):
While I appreciate the effort @Scroker has put into this, everyone needs to understand that this is a borderline unreviewable PR. It touches EVERYTHING, it does so in over _a hundred_ commits, and there are functional, stylistic, and resource changes all mixed together across a random selection of them.

Some of the diffs are HUGE due to automated changes, yet there are important functional changes mixed in with them that have to be dug out from all the noise. My system can barely handle loading the diff views, to the point that I'm composing this review message in GEdit because trying to type it into the browser form was impossible with an input lag of several seconds. (No exaggeration.)

So, because of that, I've barely looked at any of the changes here. Mostly I focused on the most recent mega-commit, and at least for that one I'm going to have to insist that:

1. It get broken up into separate commits, one for just the `eslint --fix` changes **and no others**, then one or more commits for any changes made separate from those automated fixes.
2. Whatever repo-wide search-and-replace you did to change `@param` values needs to be undone. This PR should not be touching the `jasmine.js` test framework code or the `browser-polyfill.min.js.map` file, yet the last commit modifies both of those.

I'm not sure this PR can be reviewed in this state; it really needs a more organized commit history to be making such sweeping changes to nearly all of GSConnect. The desire to get the code ported to Gtk4/libAdw does not trump the need for a clean, coherent code tree built up in a series of well-defined and clearly documented functional changes.

Scroker commented:
It is a bit complicated but I could do it, if you explain how I have to organize the commits for all the changes made I create a branch starting from the main and  report the changes to create a more organized commit structure, at this point it is a shame to stop.

Scroker commented on .gitignore:
```diff
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 _build
+builddir
```
Thanks I'll try now, otherwise I'll remove it anyway

Scroker commented on data/application.css:
```diff
@@ -130,3 +130,46 @@
+ */
+.counter {
+  background: alpha(currentColor, .15);
+  border-radius: 999px;
```
You are right.

Scroker commented on src/preferences/device.js:
```diff
@@ -1261,20 +1226,20 @@ export const DeviceNavigationPage = GObject.registerClass({
         const win = Gtk.Application.get_default().get_active_window();
         const fileDialog = new Gtk.FileDialog({
-            title: "Seleziona un file",
+            title: 'Seleziona un file',
```
I need to edit it to English and add translations.

LorierMatthieu committed:
Add primary menu argument to GtkMenuButton

Scroker committed:
Applying suggested modifications

Scroker commented on data/meson.build:
```diff
@@ -52,15 +52,6 @@ if desktop_utils.found()
-  join_paths('icons', app_id + '.svg'),
-  join_paths('icons', app_id + '-symbolic.svg')],
-  install_dir: join_paths(datadir, 'icons', 'hicolor', 'scalable', 'apps')
-)
```
This is no longer needed, now the symbolic and scalable app icon is in the gresource icon theme

Scroker commented on data/metadata.json.in:
```diff
@@ -3,6 +3,7 @@
+    "shell-version": [ "48" ],
+    "url": "@PACKAGE_URL@/wiki",
+    "gettext-domain": "org.gnome.Shell.Extensions.GSConnect"
 }
```
Here:
- some Adw component is compatible only with versio 48 of gnome or later.
- gettext-domain is needed to translate the quick settings menu

Scroker commented on data/org.gnome.Shell.Extensions.GSConnect.gresource.xml:
```diff
@@ -35,44 +42,46 @@ SPDX-License-Identifier: GPL-2.0-or-later
+    <file compressed="true" preprocess="xml-stripblanks" alias="phone-vibrate-symbolic.svg">icons/phone-vibrate-symbolic.svg</file>
+    <file compressed="true" preprocess="xml-stripblanks" alias="chat-bubbles-text-symbolic.svg">icons/chat-bubbles-text-symbolic.svg</file>
+  </gresource>
 </gresources>
```
With this change the application automatically recognize the icon theme inside the gresouce

Scroker committed:
Restored the destroy chain and create a new one for new ApplicationWindows

Scroker committed:
Trying to minimize the difference with the main branch

Scroker committed:
Trying to minimize the difference with the main branch

Scroker committed:
Trying to minimize the difference with the main branch

Scroker committed:
Try to solve the errors in testClipboardComponent.test

Scroker commented:
I don't know if I messed things up, but I tried as much as possible to eliminate unnecessary changes and reduced the changes by about ten files. I understand that the commits are very messed up, if it's okay with you I'll move the changes back to another branch with one of the more structured commits and resend the pull request using that. 

I also apologize for all the mess about destroy dispose and finalize. I did not really understand the resource deallocation mechanism of this application. Now I have understood it and tried to restore it and implemented it for new windows as well. Now I simply have a recurring error if I turn off the daemon after opening the message window:
```
../gobject/gsignal.c:2699: instance '0x556b5097d630' has no handler with id '3651'
```
Anyway I saw the errors related to the tests, I think I reproduced them locally. On the clipboard I had to make some adjustments for Gtk4 even though I have been using it for days and it was working. Anyway, the test result is now this:
```
1..2
not ok 1 The Clipboard component pulls changes from the session clipboard
# Message: Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7031)
# Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34

(gjs:551912): Gjs-WARNING **: 13:26:00.411: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
id<@file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:49:58
set text@file:///usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/components/clipboard.js:66:29
@file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:57:9
attempt@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7054:24
QueueRunner.prototype.run@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7087:25
runNext@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7006:18
next@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7013:11
once/<@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:6907:11
QueueRunner.prototype.run/<@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7103:12
setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34

not ok 2 The Clipboard component pushes changes to the session clipboard
# Message: Expected '83e51d23-bc25-4b04-8dd5-cfd32be078cf' to be '88ca6802-5cf2-4780-9be3-7f2923d30d77'.
# Stack:
#   error properties: Object({ matches: Function })
#   id<@file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:38:36
#   @file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:57:9
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
# Message: Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7031)
# Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
nome/gjs/modules/core/overrides/GLib.js:263:34
# Message: Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7031)
# Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
```
On the SMS part the real problem is that I imported your WIP: No-cache SMS support which solves a number of issues that also plague the current stable version on reading the message history.As a result of the changes made, however, the whole chain of sending the message and reading the history has changed and the test fails:
```
1..8
ok 1 The sms plugin can be loaded
ok 2 The sms plugin enables its GActions when connected
ok 3 The sms plugin requests messages when connected
not ok 4 The sms plugin can request a list of conversations
# Message: Error: <spyOn> : _handleDigest() method does not exist\nUsage: spyOn(<object>, <methodName>) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7947)
# Stack:
#   error properties: Object({ matches: Function })
#   @file:///usr/libexec/installed-tests/gsconnect/plugins/testSmsPlugin.js:226:14
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
not ok 5 The sms plugin can request full conversations
# Message: Error: <spyOn> : _handleDigest() method does not exist\nUsage: spyOn(<object>, <methodName>) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7947)
# Stack:
#   error properties: Object({ matches: Function })
#   @file:///usr/libexec/installed-tests/gsconnect/plugins/testSmsPlugin.js:235:14
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
not ok 6 The sms plugin only requests new or updated converations
# Message: Error: <spyOn> : _handleDigest() method does not exist\nUsage: spyOn(<object>, <methodName>) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7947)
# Stack:
#   error properties: Object({ matches: Function })
#   @file:///usr/libexec/installed-tests/gsconnect/plugins/testSmsPlugin.js:252:14
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
ok 7 The sms plugin can send SMS messages
ok 8 The sms plugin disables its GActions when disconnected
```

Scroker committed:
Eslint test fixes

Scroker commented:
I don't know if I messed things up, but I tried as much as possible to eliminate unnecessary changes and reduced the changes by about ten files. I understand that the commits are very messed up, if it's okay with you I'll move the changes back to another branch with one of the more structured commits and resend the pull request using that. 

I also apologize for all the mess about destroy dispose and finalize. I did not really understand the resource deallocation mechanism of this application. Now I have understood it and tried to restore it and implemented it for new windows as well. Now I simply have a recurring error if I shut down the daemon after opening the message window:
```
../gobject/gsignal.c:2699: instance '0x55ed27a5f620' has no handler with id '1932'
```
Anyway I saw the errors related to the tests, I think I reproduced them locally. On the clipboard I had to make some adjustments for Gtk4 even though I have been using it for days and it was working. Anyway, the test result is now this:
```
1..2
not ok 1 The Clipboard component pulls changes from the session clipboard
# Message: Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7031)
# Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34

(gjs:551912): Gjs-WARNING **: 13:26:00.411: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
id<@file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:49:58
set text@file:///usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/components/clipboard.js:66:29
@file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:57:9
attempt@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7054:24
QueueRunner.prototype.run@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7087:25
runNext@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7006:18
next@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7013:11
once/<@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:6907:11
QueueRunner.prototype.run/<@file:///usr/libexec/installed-tests/gsconnect/jasmine.js:7103:12
setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34

not ok 2 The Clipboard component pushes changes to the session clipboard
# Message: Expected '83e51d23-bc25-4b04-8dd5-cfd32be078cf' to be '88ca6802-5cf2-4780-9be3-7f2923d30d77'.
# Stack:
#   error properties: Object({ matches: Function })
#   id<@file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:38:36
#   @file:///usr/libexec/installed-tests/gsconnect/components/testClipboardComponent.js:57:9
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
# Message: Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7031)
# Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
nome/gjs/modules/core/overrides/GLib.js:263:34
# Message: Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7031)
# Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
Stack:
#   error properties: Object({ matches: Function })
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
```
On the SMS part the real problem is that I imported your WIP: No-cache SMS support which solves a number of issues that also plague the current stable version on reading the message history. As a result of the changes made, however, the whole chain of sending the message and reading the history has changed and the test fails:
```
1..8
ok 1 The sms plugin can be loaded
ok 2 The sms plugin enables its GActions when connected
ok 3 The sms plugin requests messages when connected
not ok 4 The sms plugin can request a list of conversations
# Message: Error: <spyOn> : _handleDigest() method does not exist\nUsage: spyOn(<object>, <methodName>) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7947)
# Stack:
#   error properties: Object({ matches: Function })
#   @file:///usr/libexec/installed-tests/gsconnect/plugins/testSmsPlugin.js:226:14
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
not ok 5 The sms plugin can request full conversations
# Message: Error: <spyOn> : _handleDigest() method does not exist\nUsage: spyOn(<object>, <methodName>) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7947)
# Stack:
#   error properties: Object({ matches: Function })
#   @file:///usr/libexec/installed-tests/gsconnect/plugins/testSmsPlugin.js:235:14
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
not ok 6 The sms plugin only requests new or updated converations
# Message: Error: <spyOn> : _handleDigest() method does not exist\nUsage: spyOn(<object>, <methodName>) in file:///usr/libexec/installed-tests/gsconnect/jasmine.js (line 7947)
# Stack:
#   error properties: Object({ matches: Function })
#   @file:///usr/libexec/installed-tests/gsconnect/plugins/testSmsPlugin.js:252:14
#   setTimeout/source<@resource:///org/gnome/gjs/modules/esm/_timers.js:72:9
#   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:263:34
ok 7 The sms plugin can send SMS messages
ok 8 The sms plugin disables its GActions when disconnected
```
Let me know how to proceed, whether to change branches and bring back the changes or continue on this one. And if you have any advice on how to deal with these issues on testing.

PS: I know that in general I shouldn't have touched the tests, but clipboard had direct dependencies with Gtk3 so I was forced, also on this let me know if it's ok.

Scroker committed:
share and notification fixes

Scroker committed:
Merge branch 'main' into gtk4-port

LorierMatthieu committed:
Give all available space to AdwSpinner

LorierMatthieu committed:
Update french translations

LorierMatthieu committed:
Change buttons' order on messaging window

Scroker committed:
Fixed the Encryption Info AlertDialog  and CommandList on device preferences page

Scroker committed:
Merge branch 'main' into gtk4-port

LorierMatthieu commented:
What's happening here?

Scroker committed:
Merge branch 'main' into gtk4-port

LorierMatthieu commented:
Is this a response? If yes, so the work is still progressing.

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
BugFix: check if address in contacts exist for group messages

Scroker committed:
Merge branch 'main' into gtk4-port

ttmx commented:
Hey! I tried this PR and its looking really nice! I am certain I have not done super thorough testing, but for now the only issue is that I cannot open the Mobile Settings menu unless I remove these lines from src/preferences/service.js
```js
        this.settings.bind(
            'missing-openssl',
            this.infobar_openssl,
            'reveal-child',
            Gio.SettingsBindFlags.DEFAULT
        );
        this.add_action(this.settings.create_action('missing-openssl'));
```

and that icons seem to be missing.
       
I have very little experience in gnome dev and I can't even seem to figure out how to log stuff, so I'm having trouble debugging it.
If you wish to give me any pointers on what is missing on this PR I would appreciate it! I have some spare time for a couple weeks.

Regardless, I will keep using this branch (with those lines removed), thank you a lot for your work, hope it gets merged soon!

LorierMatthieu commented:
Maybe [that link](https://gjs.guide/extensions/development/debugging.html) could help you debugging the extension. 

And I also noticed that link sharing from device doesn't open web browser with that link and that messaging sometimes fails to load messages even the device is connected.

ttmx committed:
Correct spelling mistake (Partecipants -> Participants)

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Migrated Gio.DesktopApplicationInfo to GioUnix.DesktopApplicationInfo and inibithed the openssl infobar, infobar redesign is needed

Scroker committed:
Bugfix: unpair device does not chagne the gsettings device schema

Scroker committed:
create the OpenSSLAlertDialog template

Scroker committed:
Implemented the OpenSSLAlertDialog widget if missing-openssl is true

LorierMatthieu commented:
@Scroker It seems I no longer have write access, can you add me again please? I have a lot of work locally.

ttmx commented:
> @LorierMatthieu , I understand, but I'm also working on new features, starting from gtk4-port branch. To avoid misalignments, I'd prefer you to fork my branch and, when you have fixes or a new UI design, send me a pull request. This would help me evaluate the impact of the changes on the other branches before accepting them, instead of having to find the changes directly on the gtk4-port branch.

https://github.com/Scroker/gnome-shell-extension-gsconnect/pull/3
Is this format okay?
Thx for your work!

Scroker committed:
Merge pull request #3 from ttmx/gtk4-translation-fix

Correct spelling mistake (Partecipants -> Participants)

Scroker committed:
Fix missing GioUnix dependency in notification service component

Scroker committed:
Correct spelling mistake pt. 2 (Partecipants -> Participants)

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Fixed show error message not displayed

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
fixed eslint errors and warnings

olumolu commented:
what is the status of this

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Migrate the porting into the new gsconnect-preference.js file

Scroker committed:
Try to fix SMS plugin test with new sms implementation

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
Merge branch 'gtk4-port' of github.com:Scroker/gnome-shell-extension-gsconnect into gtk4-port

daniellandau commented on src/preferences/keybindings.js:
```diff
@@ -84,63 +89,65 @@ export const ShortcutChooserDialog = GObject.registerClass({
-        // TODO: Critical: 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed
+        // Convertiamo il valore del tasto in minuscolo
+        let keyvalLower = Gdk.keyval_to_lower(keyval);
+        // Usiamo il state fornito, mascherando solo i bit di modificatore validi
```
I don't speak Italian, please use only English for comments and code

daniellandau commented on src/service/plugins/findmyphone.js:
```diff
@@ -181,27 +196,44 @@ const Dialog = GObject.registerClass({
+
         // Stop the alarm
-        this.cancellable.cancel();
+        this._cancellable.cancel();
```
This is a good example of a change that seems spurious. What's the reason for this change? Even digging into the commit log it's mixed in with other changes.

daniellandau commented on src/service/plugins/sms.js:
```diff
@@ -148,7 +148,7 @@ const SMSPlugin = GObject.registerClass({
         super._init(device, 'sms');
 
-        this.cacheProperties(['_threads']);
+        // this.cacheProperties(['_threads']);
```
Either this is still needed or it's not, leaving it commented out is not the way to go

daniellandau commented on src/service/plugins/sms.js:
```diff
@@ -269,40 +226,36 @@ const SMSPlugin = GObject.registerClass({
      * @param {object[]} messages - A list of sms message objects
      */
-    _handleMessages(messages) {
+    _handleDigest(messages) {
```
Why is the method name changing and the old method with that name going away?

daniellandau commented on src/service/ui/contacts.js:
```diff
@@ -530,62 +310,42 @@ export const ContactChooser = GObject.registerClass({
-        }
-
-        return false;
+        const re = new RegExp(this.search_entry.text, 'i');
```
This change has nothing to with GTK3->GTK4

daniellandau commented on src/service/ui/messaging.js:
```diff
@@ -375,7 +452,9 @@ const Conversation = GObject.registerClass({
         // each address carried by the message.
-        this._addresses = addresses;
+        this._addresses = addresses.filter((elemento, indice, self) => {
+            return self.indexOf(elemento) === indice;
```
Why add a dedup step partially in italian? Again no relation to GTK4

daniellandau commented on src/service/ui/mousepad.js:
```diff
@@ -91,243 +107,159 @@ export const InputDialog = GObject.registerClass({
         }
 
-        this.device.sendPacket({
+        const pack = {
```
Avoid adding noise like this on such a large pull request

daniellandau commented on src/service/ui/service.js:
```diff
@@ -23,7 +24,7 @@ OS:        ${GLib.get_os_info('PRETTY_NAME')}
 
 
-/**
+/*
```
I'm not marking all of these, but it's a general pattern of many changes being very unnecessary and not improvements

daniellandau commented on src/service/ui/service.js:
```diff
@@ -46,24 +47,22 @@ export const DeviceChooser = GObject.registerClass({
-            application: Gio.Application.get_default(),
-        });
-        this.set_keep_above(true);
+    Singals: {
```
This is a typo. What's the effect here of not actually having Signals defined?

daniellandau commented on webextension/js/browser-polyfill.min.js.map:
```diff
@@ -1 +1 @@
-{"version":3,"sources":["browser-polyfill.js"],"names":["Object","extensionAPIs","apiMetadata","constructor","items","get","isThenable","value","makeCallback","promise","metadata","callbackArgs","pluralizeArguments","numArgs","wrapAsyncFunction","args","minArgs","name","length","maxArgs","target","resolve","reject","console","wrapMethod","apply","wrapper","hasOwnProperty","Function","wrapObject", ...
+{"version":3,"sources":["browser-polyfill.js"],"names":["Object","extensionAPIs","apiMetadata","constructor","items","get","isThenable","value","makeCallback","promise","metadata","callbackArgs","pluralizeArguments","numArgs","wrapAsyncFunction","args","minArgs","name","length","maxArgs","target","resolve","reject","console","wrapMethod","apply","wrapper","hasOwnProperty","Function","wrapObject", ...
```
This seems to be an accidental integer->Number find-and-replace. Please don't commit stuff like this

daniellandau commented on data/icons/phonelink-lock-symbolic.svg:
```diff
@@ -1,93 +1,2 @@
-     id="layer12" />
-</svg>
+<?xml version="1.0" encoding="UTF-8"?>
+<svg xmlns="http://www.w3.org/2000/svg" height="16px" viewBox="0 0 16 16" width="16px"><g fill="#222222"><path d="m 4.195312 0 c -1.199218 0 -2.195312 1 -2.195312 2.199219 v 1.808593 h 2 v -1.808593 c 0 -0.128907 0.074219 -0.199219 0.195312 -0.199219 h 7.609376 c 0.125 0 0.195312 0.070312 0.195312 0.199219 v 9.613281 c 0 0.128906 -0.070312 0.199219 -0.195312 0.199219 l -9.804688 -0.011719 v 1.800 ...
```
What's the point of this SVG change?

daniellandau reviewed (changes requested):
By necessity this is a large pull request, but it is much much larger than it needs to be. Please leave out everything that isn't related to the Gtk migration. We can discuss some of the other changes separately, but not as part of the Gtk migration

Scroker committed:
ui: port resources and templates to GTK4

Move the GtkBuilder templates and styling to GTK4/Adw, register the new resources, update icon fallbacks, and drop unused legacy image assets.

Scroker committed:
preferences: port windows to GTK4 and Adw

Move the preferences application and device settings UI code onto GTK4/Adw APIs and the new templates.

Scroker committed:
service: port application windows to GTK4

Move the service application and shared service windows to GTK4/Adw APIs.

Scroker committed:
components: update clipboard support for GTK4

Use Gdk.Clipboard APIs, update related component imports, and adapt the clipboard tests to the GTK4 async clipboard API.

Scroker committed:
plugins: port service dialogs to GTK4

Adapt Find My Phone, Mousepad, Notification reply, and Share dialog code to the GTK4/Adw templates and APIs.

Scroker committed:
sms: port messaging UI to GTK4

Adapt the SMS plugin, conversation views, legacy messaging dialog, and contact chooser to the GTK4/Adw messaging templates while keeping the original thread-cache behavior.

Scroker committed:
metainfo: update screenshots and Italian strings

Replace the legacy AppStream screenshot with current GTK4 screenshots and include the matching Italian translation updates.

Scroker committed:
core: serialize channel packet writes

Scroker committed:
sftp: open mounted device paths

Scroker committed:
shell: keep Quick Settings submenus open

Scroker committed:
preferences: render nested device action menus

Scroker committed:
preferences: fix shortcut editor state handling

Scroker committed:
Merge branch 'main' into gtk4-port

Scroker committed:
sftp: reset mounted menu when disconnected

Scroker committed:
cleanup: remove unrelated review noise

Scroker committed:
contacts: avoid duplicate chooser rows
